lib/wor/authentication/controllers/sessions.rb

```diff
@@ -0,0 +1,120 @@
+module Wor
+  module Authentication
+    module Controllers
+      module Sessions
+        def create
+          entity = authenticate_entity(authentication_params)
+          return render_error('Invalid authentication credentials', :unauthorized) unless entity
+          render json: token_creation_response(entity), status: :created
+        rescue ActionController::ParameterMissing
+          render_error("Parameter 'session' missing.", :bad_request)
+        end
+
+        def renew
+          if decoded_token.able_to_renew?(current_entity.try(:renew_id))
+            render json: token_creation_response(current_entity), status: :created
+          else
+            render_error('Invalid renew_id', :unauthorized)
+          end
+        end
+
+        def invalidate_all
+          update_custom_validation_value(current_entity)
+          update_renew_id(current_entity)
+          head :ok
+        end
+
+        ##########################################################################################
+        #                   DEFAULT METHOD IMPLEMENTATIONS, USER SHOULD CHANGE                   #
+        ##########################################################################################
+        def authenticate_entity(params)
+          entity = User.find_by(email: authentication_params[:email])
+          return nil unless entity.present? && entity.valid_password?(authentication_params[:password])
+          entity
+        end
+
+        def entity_payload(entity)
+          { id: entity.id }
+        end
+
+        def new_token_expiration_date
+          Wor::Authentication.expiration_minutes.minutes.from_now.to_i
+        end
+
+        def new_token_renew_date
+          Wor::Authentication.renew_minutes.minutes.from_now.to_i
+        end
+
+        def token_renew_id(entity)
+          if entity.respond_to?(:renew_id)
+            current_key = entity.renew_id
+            return current_key if current_key
+            update_renew_id(entity)
+          end
+        end
+
+        def update_custom_validation_value(entity)
+          SecureRandom.hex(32).tap do |random_value|
+            begin
+              entity.update!(entity_custom_validation: random_value)
```

Comment:
Why don't you just use `entity.update!(entity_custom_validation: SecureRandom.hex(32))` ?